### ISSUE TITLE
Delayed sending, aka cool down period

### ISSUE THREAD

**Issue by sphh** [FossifyOrg/Messages#61]
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Messages/issues) or [closed](https://github.com/FossifyOrg/Messages/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Messages/discussions) or [closed](https://github.com/FossifyOrg/Messages/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Messages/blob/master/CONTRIBUTING.md).
- [ ] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

After hitting the send button, the message should only be sent after a (configurable) delay has elapsed. During this delay a count-down timer showing the remaining time should replace the send button. A circular progress meter in icon size running towards zero would be ideal. As long as the timer has not elapsed, it is possible to cancel the sending of the message, e.g. by clicking on that progress meter or pressing the back button.

If the user decides to cancel, s/he will be taken back to the message and edit it.

### Why do you want this feature?

Often you click on the send button but immediately regret having sent the message. In this case a traditional confirmation, if one really wants to send the message does not help, because this question is generally automatically confirmed.

Such a ‘cool down’ period could help you avoiding stupid/embarrassing/(or even) costly mistakes!

### Additional information

 [QKSMS](https://github.com/moezbhatti/qksms) (now [QUIK](https://github.com/octoshrimpy/quik)) solves this problem by having a (customizable) ‘cool down’ period: A countdown timer is displayed after hitting the send button. Without any interaction, the text message is dispatched when the timer reaches zero. The sending can be cancelled by the user by clicking on this countdown timer. S/he returns immediately to the text message and can edit it.

**Comment by Saturn745** [comment]
Would love to see this feature.

**Comment by SilverPuppy** [comment]
I'll agree that this is a good idea as long as it's optional, or can be set to zero. I wouldn't use it myself.

**Comment by sphh** [comment]
I agree, as mentioned it should definitely be adjustable and optional.

**Comment by Aga-C** [comment]
> Also a nice feature would be to program sending a message at defined time (a different feature from the one mentionned above).

It's already in the app.

**Comment by CaseOf** [comment]
> > Also a nice feature would be to program sending a message at defined time (a different feature from the one mentionned above).
> 
> It's already in the app.

awesome!
